src/app/shared/calendar.component.ts

```diff
@@ -85,7 +85,7 @@ export class PlanetCalendarComponent implements OnInit {
         } : {
           backgroundColor: styleVariables.accent, borderColor: styleVariables.accent, textColor: styleVariables.accentText
         };
-        return this.eventObject({ ...task, isTask: true }, task.deadline, task.deadline, taskColors);
+        return this.eventObject({ ...task, isTask: true }, task.deadline, task.completed ? task.completedTime: '', taskColors);
```

Comment:
The third parameter should remain `task.deadline`.  That parameter only changes how the task is displayed on the calendar view, and if the task is completed on a day after the deadline this will show the task spanning multiple days, which is not desired.